### PR TITLE
JS1 Week4 HW

### PR DESCRIPTION
**Pull request by alina-kho**

### CODE REVIEW

File: javascript/javascript1/week4/voice-assistant.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
+"use strict";
+
+//Creating global scope variables
+let userName = "";
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js:4]: Is there any reason to make this a global scope?

File: javascript/javascript1/week4/voice-assistant.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+let todos = []; //declared through let to be able to modify the array through the filter method
+
+//Creating a function
+function getReply(command) {
+  let commandSplitted = command.split(" "); //Splits the command into separate words for easy search through them
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js]: Could this be a const?

File: javascript/javascript1/week4/voice-assistant.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+
+  //Name actions
+  //.toLowerCase is used to get rid of case sensitivity
+  if (
+    command.toLowerCase().includes("my name is") ||
+    command.toLowerCase().includes("i am")
+  ) {
+    let name = commandSplitted[commandSplitted.length - 1];
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js]: This is not reassigned, so could also be const

File: javascript/javascript1/week4/voice-assistant.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+let todos = []; //declared through let to be able to modify the array through the filter method
+
+//Creating a function
+function getReply(command) {
+  let commandSplitted = command.split(" "); //Splits the command into separate words for easy search through them
+
+  //Name actions
+  //.toLowerCase is used to get rid of case sensitivity
+  if (
+    command.toLowerCase().includes("my name is") ||
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js:14]: Great technic to use toLowerCase

File: javascript/javascript1/week4/voice-assistant.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+    } else {
+      userName = name;
+      return `Nice to meet you ${name}`;
+    }
+  } else if (command.toLowerCase().includes("is my name")) {
+    if (userName == undefined || userName == "") {
+      return "I haven't met you yet. Please introduce yourself";
+    } else {
+      return userName;
+    }
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js:29]: This works great, but you could use the "Return Early" technic. (no need for the last else statement)
If you have time, you could take a look at it: https://dev.to/jenniferlynparsons/early-returns-in-javascript-5hfb

File: javascript/javascript1/week4/voice-assistant.js [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ]; // array created to change a number value to a word
+    let today = new Date(); //getting today's date
+    let day = today.getDate();
+    let month = months[today.getMonth()]; //today.getMonth() will give us a value from 0 to 11, so we can easily apply it to find a month in the months array
+    let year = today.getFullYear();
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js]: Could these be consts?

File: javascript/javascript1/week4/voice-assistant.js [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+  else if (
+    command.toLowerCase().includes("prediction") ||
+    command.toLowerCase().includes("predict") ||
+    command.toLowerCase().includes("tell my fortune")
+  ) {
+    let randomNum = Math.floor(Math.random() * 6);
+    switch (randomNum) {
+      case 0:
+        return `You are going to have an amazing day!`;
+        break;
```
Comment (Kristoffer88) [inline javascript/javascript1/week4/voice-assistant.js]: I do not think break is nessecary, as you are returning right above it.

**Review by Kristoffer88 — APPROVED** [review]
Overall really good.

You might want to look into how to avoid nesting the if/else-if/else statement. To make it a little bit more readable. This is a advanced topic, so only if you have time to look into it

File: javascript/javascript1/week4/voice-assistant.js (hunk 1 above)
Comment (alina-kho) [reply javascript/javascript1/week4/voice-assistant.js:4]: Thanks a lot for the feedback. I fixed variable declaration (let -> const) in the cases you mentioned - everything works perfectly. I also deleted breaks in the switch statement as you said - I don't need them if I have returns. 

As for userName as a global scope, I tried moving it inside the function, but some of the features (1 - when a user says his name again and the assistant answers with "Hi again"; 2 - when a user asks for the name) stopped working. Maybe there is a way out of it.

File: javascript/javascript1/week4/voice-assistant.js (hunk 5 above)
Comment (alina-kho) [reply javascript/javascript1/week4/voice-assistant.js:29]: Thanks! I will look into it :)

File: javascript/javascript1/week4/voice-assistant.js (hunk 2 above)
Comment (alina-kho) [reply javascript/javascript1/week4/voice-assistant.js]: Fixed.

File: javascript/javascript1/week4/voice-assistant.js (hunk 3 above)
Comment (alina-kho) [reply javascript/javascript1/week4/voice-assistant.js]: Fixed.

File: javascript/javascript1/week4/voice-assistant.js (hunk 6 above)
Comment (alina-kho) [reply javascript/javascript1/week4/voice-assistant.js]: Fixed.

File: javascript/javascript1/week4/voice-assistant.js (hunk 7 above)
Comment (alina-kho) [reply javascript/javascript1/week4/voice-assistant.js]: Fixed

**Pull request merged**